extensions/amp-a4a/0.1/test/test-a4a-integration.js

```diff
@@ -249,7 +249,7 @@ describe('integration test: a4a', () => {
     const forceCollapseStub =
```

Comment:
Added a check to ensure that forceCollapseStub is not called when xhr rejects. All other tests that you mention already exist, except for the empty body check which is being added in PR #7334.